### PR TITLE
Added hash map tests for Ruby

### PR DESCRIPTION
**Pull request by elishamutang**
## Because

Allows users to populate their hash maps where it will exceed their load factor and ensuring that their hash map methods will still work as expected after growing their buckets ([similar to this PR for JavaScript course](https://github.com/TheOdinProject/curriculum/pull/28245)).

## This PR

Added a new section to test out user hash map implementation after `Assignment` section.
Added dummy nodes to populate user hash map in the new section.

## Issue

Closes [#28257](https://github.com/TheOdinProject/curriculum/issues/28257)

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by SumonGFC** [comment]
Hello :p Small gripe: In ruby style conventions, the names of "predicate methods" (methods that return a boolean) should end with a question mark to indicate that they return booleans. So on lines 52 and 109 in your proposed changes, `#has(key)` should be named `#has?(key)`. just wanted to let you know.

**Comment by elishamutang** [comment]
@SumonGFC Ah i see, didn't know that. Thanks for letting me know, I'll undo that change! What about the the set method in lines 44 & 48?

**Comment by SumonGFC** [comment]
@elishamutang those lines are perfect. This set method is not supposed to return a boolean in the interface described by this project (think it is supposed to return the value that is associated with the key)

File: ruby/computer_science/project_hash_map.md [hunk 1]
```diff
@@ -63,6 +65,49 @@ You already know the magic behind hash maps, now it's time to write your own imp
...
+
+    ```ruby
+    test.set('moon', 'silver')
+    ```
+
+1. If you have implemented your hash map correctly, the capacity of your new hash map will drop well below your load factor and you will notice that the nodes in your hash map are spread much evenly among your buckets.
+
+1. With your new hash map, try overwriting a few nodes using `#set(key, value)`. Again, this should only over-write existing `values` of your nodes.
+
+1. Test the other methods of your hash maps such as `#get(key)`, `#has(key)`, `#remove(key)`, `#length`, `#clear`, `#keys`, `#values`, and `#entries` to check if they are still working as expected after expanding your hash map.
```
Comment (JoshDevHub) [inline ruby/computer_science/project_hash_map.md:109]:
```suggestion
1. Test the other methods of your hash maps such as `#get(key)`, `#has?(key)`, `#remove(key)`, `#length`, `#clear`, `#keys`, `#values`, and `#entries` to check if they are still working as expected after expanding your hash map.
```

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Awesome work :raised_hands:

Like @SumonGFC said, the only suggestion I have is to append a `?` to the `#has` method name (I forgot to mention this as something to keep in mind).

After that's fixed, we should be good to merge.

**Comment by elishamutang** [comment]
@JoshDevHub totally missed that last `#has` method haha. No worries and thanks for the help! You too @SumonGFC :)

**Review by JoshDevHub — APPROVED** [review]
Thanks for doing this!

**Pull request merged**